test/e2e/example_verifier.go

```diff
@@ -56,3 +62,83 @@ func verifyJbpmSpringBootExample(t *testing.T, kogitoService *v1alpha1.KogitoApp
 	}
 	assert.Equal(t, 200, resp.StatusCode)
 }
+
+func verifyJbpmQuarkusPersistentExample(t *testing.T, f *framework.Framework, kogitoService *v1alpha1.KogitoApp) {
+	// Create an order
+	orderContent := strings.NewReader("{\"approver\" : \"john\", \"order\" : {\"orderNumber\" : \"12345\", \"shipped\" : false}}")
+	resp, err := http.Post(kogitoService.Status.Route+"/orders", "application/json", orderContent)
+	if err != nil {
+		verifierLog.Fatalf("Error while trying to get the application endpoint: %s", err)
+	}
+	assert.Equal(t, 200, resp.StatusCode)
+
+	// Scale to 0 and back to 1 to spin up new application pod
+	scaleKogitoService(f, kogitoService, 0)
+	scaleKogitoService(f, kogitoService, 1)
+
+	// Check that order is still available
+	resp, err = http.Get(kogitoService.Status.Route + "/orders")
+	if err != nil {
+		verifierLog.Fatalf("Error while trying to get the application endpoint: %s", err)
+	}
+	assert.Equal(t, 200, resp.StatusCode)
+
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		verifierLog.Fatalf("Error while trying to read application response body: %s", err)
+	}
+	assert.NotNil(t, body)
+	assert.NotEmpty(t, body)
+	assert.Contains(t, string(body), "\"orderNumber\":\"12345\"")
+	assert.Contains(t, string(body), "\"id\"")
+}
+
+func scaleKogitoService(f *framework.Framework, kogitoApp *v1alpha1.KogitoApp, replicas int) {
+	verifierLog.Infof("Scaling Kogito application %s to %d replicas", kogitoApp.Name, replicas)
+
+	// Load actual Kogito application state
+	f.Client.Get(goctx.TODO(), types.NamespacedName{Name: kogitoApp.Name, Namespace: kogitoApp.Namespace}, kogitoApp)
+
+	// Adjust number of replicas
+	convertedReplicas := int32(replicas)
+	kogitoApp.Spec.Replicas = &convertedReplicas
+
+	// Update Kogito application
+	if err := f.Client.Update(goctx.TODO(), kogitoApp); err != nil {
+		verifierLog.Fatalf("Error while trying to update Kogito application: %s", err)
+	}
+
+	//wait until desired number of replicas are available
+	dc := appsv1.DeploymentConfig{}
+	for i := 1; i <= 60; i++ {
```

Comment:
Maybe we could factorize those waiting loops to have cleaner code

something like `waitFor(fn function(interface{}) error)` or similar ?